docs/dev/ui_description.rst

```diff
@@ -0,0 +1,259 @@
+Glupyter Framework Overview
+===========================
+
+The glue-jupyter ("glupyter") package supports interacting with and
+visualizing data within the Jupyter environment using core elements from
+the Glue python package. It is distinct because unlike the more
+prominent Glue package, glupyter does not leverage Qt as the front-end
+GUI library. Instead, glupyter maintains the separation between the data
+model (e.g. the core elements from Glue that are not dependent on the
+front-end library), and the view of the data (in this case, web-based
+tooling provided by Jupyter).
+
+Glupyter implements a base ``JupyterApplication`` object through which
+users can manage their data, create viewers, and add links between data
+sets. The data management and linking are controlled separate from the
+viewers in that changes made directly to the data state propagate to the
+UI -- that is, the UI does not contain any state, which allows the application to be easily controlled from code. The viewers themselves are based on the
+IPyWidget package which allows the creation of widgets that can be used
+and interacted with in python, but rendered in a javascript environment.
```

Comment:
would it be more correct to say a 'browser' environment? (e.g. not command-line npm)